test/spec/functional/listView.js

```diff
@@ -398,6 +404,84 @@ describe('A List View', function() {
+      threshold = 500;
+    for (i = 0; i < numberOfViews; i++) {
+      models.push(new Model())
+    }
+    myListView.dispose();
+    myListView = new MyListView({
+      collection: myCollection,
+      childModel: 'item',
+      childView: ItemView
+    });
+    myListView.attach($('body'));
+    myCollection.reset(models);
+    startTime = new Date().getTime();
+    myCollection.reset(models.slice(0, numberOfViews - 100));
+    endTime = new Date().getTime() - startTime;
+    console.log('Removed 100 old views at the end of ' + numberOfViews + ' views using reset in ' + endTime + 'ms');
+    expect(myListView.getItemViews().length).toBe(numberOfViews - 100);
+    expect(endTime < threshold).toBe(true);
+
+    startTime = new Date().getTime();
+    myCollection.reset(models.slice(100, numberOfViews - 100));
+    endTime = new Date().getTime() - startTime;
+    console.log('Removed 100 old views at the front of ' + (numberOfViews - 100) + ' views using reset in ' + endTime + 'ms');
+    expect(myListView.getItemViews().length).toBe(numberOfViews - 200);
+    expect(endTime < threshold).toBe(true);
+
+    var newModels = []
+    for (i = 0; i < 100; i++) {
+      newModels.push(new Model());
+    }
+    newModels = newModels.concat(models.slice(100, numberOfViews - 100));
+    startTime = new Date().getTime();
+    myCollection.reset(newModels);
+    endTime = new Date().getTime() - startTime;
+    console.log('Added 100 new views at the front of ' + (numberOfViews - 200) + ' views using reset in ' + endTime + 'ms');
+    expect(myListView.getItemViews().length).toBe(numberOfViews - 100);
+    expect(myListView.getItemViews()[0].model.cid).toBe(myCollection.at(0).cid);
+    expect(myListView.getItemViews()[numberOfViews - 101].model.cid).toBe(myCollection.at(numberOfViews - 101).cid);
+    expect(endTime < threshold).toBe(true);
+
+    threshold = 1000;
+
+    myListView.dispose();
+    myListView = new MyListView({
+      collection: myCollection,
+      childModel: 'item',
+      childView: ItemView,
+      emptyTemplate: Handlebars.compile('<div class="empty-list"></div>'),
+      template: Handlebars.compile('<div class="templated-list"></div><div inject="children"></div>'),
+      childrenContainer: 'children'
+    });
+    myListView.attach($('body'));
+    myCollection.reset(models);
+    startTime = new Date().getTime();
+    myCollection.reset(newModels);
+    endTime = new Date().getTime() - startTime;
+    console.log('A list view using a children container: added 100 to the front and removed 200 views to ' + numberOfViews + ' item views using reset in ' + endTime + 'ms');
+    expect(myListView.getItemViews().length).toBe(numberOfViews - 100);
+    expect(myListView.getItemViews()[0].model.cid).toBe(myCollection.at(0).cid);
+    expect(myListView.getItemViews()[numberOfViews - 101].model.cid).toBe(myCollection.at(numberOfViews - 101).cid);
+    expect(endTime < threshold).toBe(true);
+
+    _.shuffle(models);
```

Comment:
new test case.

the test cases can share a describe and before if you want to share setup.